tests/functional/splinter_test.c

```diff
@@ -866,248 +1004,314 @@ test_splinter_perf(trunk_config    *cfg,
+   if (!SUCCESS(rc)) {
+      return rc;
+   }
+   for (uint64 i = 0; i < num_range_threads; i++) {
+      platform_thread_join(params[i].thread);
+   }
 
-      for (uint64 i = 0; i < num_range_threads; i++) {
-         if (!SUCCESS(params[i].rc)) {
-            rc = params[i].rc;
-            goto destroy_splinter;
-         }
+   uint64 total_time = platform_timestamp_elapsed(start_time);
+
+   for (uint64 i = 0; i < num_range_threads; i++) {
+      rc = params[i].rc;
+      if (!SUCCESS(rc)) {
+         return rc;
       }
+   }
 
-      rc = STATUS_OK;
+   rc = STATUS_OK;
 
-      platform_log("\nper-splinter per-thread range time per tuple %lu ns\n",
-                   total_time * num_range_threads / total_ranges);
-      platform_log("splinter total range rate: %lu ops/second\n",
-                   SEC_TO_NSEC(total_ranges) / total_time);
-      for (uint8 spl_idx = 0; spl_idx < num_tables; spl_idx++) {
-         trunk_handle *spl = spl_tables[spl_idx];
-         cache_assert_free(spl->cc);
-         trunk_print_lookup_stats(spl);
-         cache_print_stats(spl->cc);
-         cache_reset_stats(spl->cc);
-      }
+   platform_log("\nper-splinter per-thread range time per tuple %lu ns\n",
+                total_time * num_range_threads / total_ranges);
+   platform_log("splinter total range rate: %lu ops/second\n",
+                SEC_TO_NSEC(total_ranges) / total_time);
+   for (uint8 spl_idx = 0; spl_idx < num_tables; spl_idx++) {
+      trunk_handle *spl = spl_tables[spl_idx];
+      cache_assert_free(spl->cc);
+      trunk_print_lookup_stats(spl);
+      cache_print_stats(spl->cc);
+      cache_reset_stats(spl->cc);
+   }
+   platform_log("%s() completed.\n", __FUNCTION__);
+   return rc;
+}
 
-      ZERO_CONTENTS_N(curr_op, num_tables);
-      total_ranges = 0;
-      for (uint8 i = 0; i < num_tables; i++) {
-         per_table_ranges[i] =
-            ROUNDUP(per_table_ranges[i] / 4, TEST_RANGE_GRANULARITY);
-         total_ranges += per_table_ranges[i];
-      }
-      for (uint64 i = 0; i < num_range_threads; i++) {
-         params[i].total_ops      = per_table_ranges;
-         params[i].op_granularity = TEST_RANGE_GRANULARITY;
-         params[i].range_min      = 512;
-         params[i].range_max      = 1024;
-      }
+/*
+ * -----------------------------------------------------------------------------
+ * test_splinter_perf() --
+ *
+ * SplinterDB Performance exerciser. For a specified # of threads of each type,
+ * execute:
+ *  - Inserts
+ *  - Lookups
+ *  - Range queries, of 3 diff ranges { small, medium, large }
+ *
+ * We don't quite declare pass / fail based on the performance, but simply
+ * exercise this concurrent workload of different types, and report the
+ * performance metrics (such as latency, elapsed time, throughput, ...).
+ * -----------------------------------------------------------------------------
+ */
+static platform_status
+test_splinter_perf(trunk_config    *cfg,
```

Comment:
New body of refactored test-case, which shrinks quite a lot. Improves overall code readability.